### PR TITLE
[sn-platform]: update several images repo and version for CVE

### PR DESCRIPTION
**Pull request by ericsyh**
*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #<xyz>

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: #<xyz>

### Motivation

Updated the image for vault, prometheus, node_exporter and alertmanager repo and version for CVE fix, refer the scan report on https://artifacthub.io/packages/helm/prometheus-community/prometheus and https://artifacthub.io/packages/helm/hashicorp/vault

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [x] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

Check the box below.

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

**Comment by ericsyh** [comment]
@fantapsody @maxsxu @FushuWang @ciiiii @tuteng @urfreespace PTAL

File: charts/sn-platform/values.yaml [hunk 1]
```diff
@@ -230,8 +230,8 @@ images:
     repository: streamnative/pulsar_vault_init
     tag: "v1.0.6"
   custom_metric_server_prometheus:
-    repository: prom/prometheus
-    tag: "v2.39.1"
+    repository: quay.io/prometheus/prometheus
+    tag: "v2.41.0"
```
Comment (ciiiii) [inline charts/sn-platform/values.yaml:234]: > tag: "v2.24.0"
> tag: "v2.41.0"

Should we use two different versions of prometheus at the same time?

Comment (ericsyh) [reply charts/sn-platform/values.yaml:234]: No, should use same.

Comment (ericsyh) [reply charts/sn-platform/values.yaml:234]: updated in https://github.com/streamnative/charts/pull/977/commits/11c2901c4e6876c442b23524ab776ffa788bc761

**Review by maxsxu — APPROVED** [review]

**Pull request merged**